src/compiler/parser.ts

```diff
@@ -1822,20 +1824,37 @@ namespace ts {
             }
 
             let tokenPos = scanner.getTokenPos();
+            let nextCharCode: number;
+
             nextToken();
             finishNode(node);
 
-            // Octal literals are not allowed in strict mode or ES5
-            // Note that theoretically the following condition would hold true literals like 009,
-            // which is not octal.But because of how the scanner separates the tokens, we would
-            // never get a token like this. Instead, we would get 00 and 9 as two separate tokens.
-            // We also do not need to check for negatives because any prefix operator would be part of a
-            // parent unary expression.
-            if (node.kind === SyntaxKind.NumericLiteral
-                && sourceText.charCodeAt(tokenPos) === CharacterCodes._0
-                && isOctalDigit(sourceText.charCodeAt(tokenPos + 1))) {
-
-                node.flags |= NodeFlags.OctalLiteral;
+            if (node.kind === SyntaxKind.NumericLiteral) {
+                // Octal literals are not allowed in strict mode or ES5
+                // Note that theoretically the following condition would hold true literals like 009,
+                // which is not octal.But because of how the scanner separates the tokens, we would
+                // never get a token like this. Instead, we would get 00 and 9 as two separate tokens.
+                // We also do not need to check for negatives because any prefix operator would be part of a
+                // parent unary expression.
+                if (sourceText.charCodeAt(tokenPos) === CharacterCodes._0 && isOctalDigit(sourceText.charCodeAt(tokenPos + 1))) {
+                    node.flags |= NodeFlags.OctalLiteral;
+                }
+                else if (token === SyntaxKind.DotToken) {
+                    // Issue #2632 Keep space for  3 .toString()
+                    if (isWhiteSpace(sourceText.charCodeAt(node.end))) {
+						node.end++;

```

Comment:
what if the whitespace is more than one, this means you will consume the dot again. you need to set the end of the node to scanner.getTextPos() instead.